first pass at writing docs that will match the v2 branches (#3)
tablelandnetwork/docs: building/javascript-sdk.md

const tbl = await connect({ network: "testnet" });

const createRes = await tbl.create(
  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
  `name text, id int, primary key (id);`, // table schema definition
  `mytable` // optional perfix that will be added to "queryable name"
);

// `queryableName` will be the prefix, if you chose one, with the chain id and
// table id separated by underscores 
const queryableName = createRes.name;

> console.log(queryableName); // e.g. mytable_4_1

Reviewer: I can't do a "suggestion" on delete lines for some reason, so here's the suggestion:

```
// `name` will be the prefix, if supplied, with the chain id and
// table id separated by underscores.
console.log(createRes.name)
```

Now that name is much simpler to reason about, I don't think we need to have a separate variable for it here. That makes the example simpler, and highlights that they'll get the name from the response, and they should just use that.

Reviewer: The above obviously implies that we'll need to change all use of queryableName => createRes.name. The alternative is to simply destructure the create response into name and whatever else we want/need in the example for a cleaner demo?

const tbl = await connect({ network: "testnet", signer });

const createRes = await tbl.create(
  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
  `name text, id int, primary key (id);`, // table schema definition
  `mytable` // optional perfix that will be added to "queryable name"
);

> // `queryableName` will be the prefix, if you chose one, with the chain id and

Reviewer: Same deal here, keep it simple and just log the name directly.

const signer = wallet.connect(provider);

const createRes = await tbl.create(
  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
  `name text, id int, primary key (id);`, // table schema definition
  `mytable` // optional perfix that will be added to "queryable name"
);

// `queryableName` will be the prefix, if you chose one, with the chain id and
// table id separated by underscores 

> const queryableName = createRes.name;

Reviewer: you get the idea

const tbl = await connect({ network: "testnet" });

const createRes = await tbl.create(
  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
  `name text, id int, primary key (id);`, // table schema definition
  `mytable` // optional perfix that will be added to "queryable name"
);

// `queryableName` will be the prefix, if you chose one, with the chain id and
// table id separated by underscores 
const queryableName = createRes.name;

> console.log(queryableName); // e.g. mytable_4_1

Reviewer: last one

### Creating Tables

Like most relational database systems, Tableland requires the user to create tables for storing, querying, and relating data. This is done via the `create` function. The `create` function takes 3 arguments: 

> 1. The chainId of the chain you specified while connecting.

Reviewer: When we make this optional, let's also document that it is optional here. Note also that that means we have to change the structure of the function signature. We can't have an optional as the first argument.

> 2. The column names, datatypes, and constraints that would go inside the parenthesis of a full `CREATE TABLE` statement.

Reviewer: Maybe we should provide an example here?

```
`(
  id int,
  name string,
  value real
)`
```

3. An optional "prefix" to the universally unique Tableland table name.

> All tables in Tableland are unique, and the way uniqueness is ensured is that each table is named by combining the chainId of the chain the table is minted on, and the id of the ERC-721 token that represents the table.  If the prefix is specified it is added to to start of the table name.  As such the prefix must follow the rules SQL table naming, e.g it cannot start with a number and the dash character cannot be used.  For more on rules of SQL table names see [Naming Syntax](https://www.postgresql.org/docs/7.0/syntax525.htm)

Reviewer: ```suggestion
All tables in Tableland are unique, and the way uniqueness is ensured is that each table is named by combining the chainId of the chain the table is minted on, and the id of the ERC-721 token that represents the table. If the prefix is specified it is added to the start of the table name. As such the prefix must follow standard SQL table naming conventions, i.e. the prefix cannot start with a number, and only ASCII characters, numbers, and underscores may be used. For more on the rules of SQL table names see (for example) [Naming Syntax](https://www.postgresql.org/docs/7.0/syntax525.htm)
```

Also no double spaces after a period!

> Most valid SQL _constraints_ are supported, and the following data types are currently [supported](https://github.com/textileio/go-tableland/blob/main/pkg/parsing/query\_validator.go):

Reviewer: Actually, we should update this to point to the tableland spec docs now: 

```suggestion
Most valid SQL _constraints_ are supported, and the following data types are currently [supported](https://textile.notion.site/Data-Types-4831202597e04823998fe7addc275fea):
```

> * `int2`, `int4`, `int8`, `serial`, `bigserial`

Reviewer: ```suggestion
* `int`, `real`, `text`, `blob`, `any`
```

> * `text`, `uri`, `varchar`, `bpchar`

Reviewer: ```suggestion

See also the [Data Types docs](https://textile.notion.site/Data-Types-4831202597e04823998fe7addc275fea) for details on representing other common SQL data types using the above core types.
```

// Assumes a connection has already been established as above

const createRes = await tbl.create(
  4,  // Chain Id, note that since we connected to rinkeby testnet above this **MUST** be 4
  `name text, id int, primary key (id);`, // table schema definition
  `mytable` // optional perfix that will be added to "queryable name"
);

> // `queryableName` will be the prefix, if you chose one, with the chain id and

Reviewer: same again, wow we use this example a lot!

// table id separated by underscores, i.e. `${prefix}_${chainId}_${tableId}`
const queryableName = createRes.name;
console.log(queryableName); // e.g. mytable_4_1

```

> Creating a table generates a unique table identifier (table id). The table id is unique within the specified chain and can be used to reference the table for queries and updates after it has been created.  This table id is the tokenId from the Tableland Registry Smart Contract which means you can view your table on any blockchain scanner associated with the chain your table was minted on, Here is an example of a Tableland rinkeby table viewed on [Opensea](https://testnets.opensea.io/assets/0x30867ad98a520287ccc28cde70fcf63e3cdb9c3c/723)

Reviewer: ```suggestion
Creating a table generates a unique table identifier (table id). The table id is unique within the specified chain and can be used to reference the table for queries and updates after it has been created.  This table id is the tokenId from the Tableland Registry Smart Contract which means you can view your table on any blockchain scanner associated with the chain your table was minted on. Here is an example of a Tableland rinkeby table viewed on [Opensea](https://testnets.opensea.io/assets/0x30867ad98a520287ccc28cde70fcf63e3cdb9c3c/723)
```
